test/parallel/test-stringbytes-external-at-max.js

```diff
@@ -1,22 +1,33 @@
 'use strict';
+// Flags: --expose-gc
 
 require('../common');
 const assert = require('assert');
+const os = require('os');
 
 // v8 fails silently if string length > v8::String::kMaxLength
 // v8::String::kMaxLength defined in v8.h
 const kStringMaxLength = process.binding('buffer').kStringMaxLength;
 
+const skipMessage =
+  '1..0 # Skipped: intensive toString tests due to memory confinements';
+if (os.totalmem() <= 0x20000000 /* 512MB */) {
```

Comment:
Nice idea. Something like `common.kMaxTestMemUsage` or similar. Not a blocker for me, but would be useful.